### ISSUE TITLE
Multiple genres importing as a single entry

### ISSUE THREAD

**Issue by xcorvis** [jellyfin/jellyfin#12472]
### This issue respects the following points:

- [X] This is a **bug**, not a question or a configuration issue; Please visit our forum or chat rooms first to troubleshoot with volunteers, before creating a report. The links can be found [here](https://jellyfin.org/contact/).
- [X] This issue is **not** already reported on [GitHub](https://github.com/jellyfin/jellyfin/issues?q=is%3Aopen+is%3Aissue) _(I've searched it)_.
- [X] I'm using an up to date version of Jellyfin Server stable, unstable or master; We generally do not support previous older versions. If possible, please update to the latest version before opening an issue.
- [X] I agree to follow Jellyfin's [Code of Conduct](https://jellyfin.org/docs/general/community-standards.html#code-of-conduct).
- [X] This report addresses only a single issue; If you encounter multiple issues, kindly create separate reports for each one.

### Description of the bug

I manage my library with beets and used the lastgenre plugin to add multiple genres to my tracks, in the form of a genre tag, for example: "Classic Rock, Rock, Southern Rock". Jellyfin is seeing the genre string and importing the string as a _single_ genre. In the previous example, this single genre is titled "Classic Rock, Rock, Southern Rock" as opposed to the item being under 3 separate genres "Classic Rock", "Rock", "Southern Rock".
I know this used to work in jellyfin but I recently rebuilt my jellyfin server and this time around it's not picking them up as separate items.

![Screenshot 2024-08-19 131313](https://github.com/user-attachments/assets/8ca83806-b536-43dd-a4b1-45bb0664331a)

### Reproduction steps

Have an MP3 with the "genre" tag and an entry like "Classic Rock, Rock, Southern Rock" (comma and space delimited).
Create a new jellyfin library. Import the song folder. Wait
Genre for the song will be "Classic Rock, Rock, Southern Rock" instead of 3 genres that each contain the song.

### What is the current _bug_ behavior?

Songs imported with multiple-entry genre tags have those genres imported incorrectly as a single item instead of multiple items.

### What is the expected _correct_ behavior?

Genre should be detected as multiple entries and the song included under each genre.

### Jellyfin Server version

10.9.9+

### Specify the build version

10.9.9

### Environment

```markdown
- OS: Pop!_OS 22.04
- Linux Kernel: 6.9.3
- Virtualization: no
- Clients: Browser
- Browser: Firefox
- FFmpeg Version: 6.0.1-8-jammy
- Playback Method: Direct or Transcode
- Hardware Acceleration: none
- GPU Model: Nvidia
- Plugins: none
- Reverse Proxy: none
- Base URL: none
- Networking: host
- Storage: local
```

### Jellyfin logs

```shell
[2024-08-19 07:04:52.120 -05:00] [INF] Jellyfin version: "10.9.9"
[2024-08-19 07:04:52.248 -05:00] [INF] Environment Variables: ["[JELLYFIN_USER, jellyfin]", "[JELLYFIN_CONFIG_DIR, /etc/jellyfin]", "[JELLYFIN_CACHE_DIR, /var/cache/jellyfin]", 
"[JELLYFIN_DATA_DIR, /var/lib/jellyfin]", "[JELLYFIN_WEB_OPT, --webdir=/usr/share/jellyfin/web]", "[JELLYFIN_ADDITIONAL_OPTS, ]", "[JELLYFIN_LOG_DIR, /var/log/jellyfin]", "[JELLYFIN_FFMPEG_OPT, --ffmpeg=/usr/lib/jellyfin-ffmpeg/ffmpeg]", "[JELLYFIN_ARGS, $JELLYFIN_WEB_OPT $JELLYFIN_FFMPEG_OPT $JELLYFIN_SERVICE_OPT $JELLYFIN_NOWEBAPP_OPT $JELLFIN_ADDITIONAL_OPTS --datadir $JELLYFIN_DATA_DIR --configdir $JELLYFIN_CONFIG_DIR --logdir $JELLYFIN_LOG_DIR --cachedir $JELLYFIN_CACHE_DIR]"]
[2024-08-19 07:04:52.268 -05:00] [INF] Arguments: ["/usr/lib/jellyfin/bin/jellyfin.dll", "--webdir=/usr/share/jellyfin/web", "--ffmpeg=/usr/lib/jellyfin-ffmpeg/ffmpeg"]
[2024-08-19 07:04:52.268 -05:00] [INF] Operating system: "Pop!_OS 22.04 LTS"
[2024-08-19 07:04:52.268 -05:00] [INF] Architecture: X64
[2024-08-19 07:04:52.276 -05:00] [INF] 64-Bit Process: True
[2024-08-19 07:04:52.276 -05:00] [INF] User Interactive: True
[2024-08-19 07:04:52.276 -05:00] [INF] Processor count: 4
[2024-08-19 07:04:52.276 -05:00] [INF] Program data path: "/var/lib/jellyfin"
[2024-08-19 07:04:52.276 -05:00] [INF] Log directory path: "/var/log/jellyfin"
[2024-08-19 07:04:52.276 -05:00] [INF] Config directory path: "/etc/jellyfin"
[2024-08-19 07:04:52.276 -05:00] [INF] Cache path: "/var/cache/jellyfin"
[2024-08-19 07:04:52.276 -05:00] [INF] Web resources path: "/usr/share/jellyfin/web"
[2024-08-19 07:04:52.276 -05:00] [INF] Application directory: "/usr/lib/jellyfin/bin/"
[2024-08-19 07:04:53.629 -05:00] [INF] Setting cache path: "/var/cache/jellyfin"
[2024-08-19 07:04:53.865 -05:00] [INF] Loading assemblies
[2024-08-19 07:04:54.207 -05:00] [INF] Defined LAN subnets: ["127.0.0.1/8", "10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16"]
[2024-08-19 07:04:54.208 -05:00] [INF] Defined LAN exclusions: []
[2024-08-19 07:04:54.208 -05:00] [INF] Used LAN subnets: ["127.0.0.1/8", "10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16"]
[2024-08-19 07:04:54.210 -05:00] [INF] Filtered interface addresses: ["127.0.0.1", "192.168.50.2"]
[2024-08-19 07:04:54.211 -05:00] [INF] Bind Addresses ["0.0.0.0"]
[2024-08-19 07:04:54.212 -05:00] [INF] Remote IP filter is "Allowlist"
[2024-08-19 07:04:54.212 -05:00] [INF] Filtered subnets: []
[2024-08-19 07:05:09.283 -05:00] [INF] Loaded plugin: "TMDb" "10.9.9.0"
[2024-08-19 07:05:09.285 -05:00] [INF] Loaded plugin: "Studio Images" "10.9.9.0"
[2024-08-19 07:05:09.286 -05:00] [INF] Loaded plugin: "OMDb" "10.9.9.0"
[2024-08-19 07:05:09.328 -05:00] [INF] Loaded plugin: "MusicBrainz" "10.9.9.0"
[2024-08-19 07:05:09.329 -05:00] [INF] Loaded plugin: "AudioDB" "10.9.9.0"
[2024-08-19 07:05:09.497 -05:00] [INF] Kestrel is listening on "0.0.0.0"
[2024-08-19 07:05:10.523 -05:00] [WRN] The WebRootPath was not found: "/var/lib/jellyfin/wwwroot". Static files may be unavailable.
[2024-08-19 07:05:11.092 -05:00] [INF] Running startup tasks
[2024-08-19 07:05:11.149 -05:00] [INF] Daily trigger for "Generate Trickplay Images" set to fire at 2024-08-20 03:00:00.000 -05:00, which is 19:54:48.8510388 from now.
[2024-08-19 07:05:11.162 -05:00] [INF] Daily trigger for "Extract Chapter Images" set to fire at 2024-08-20 02:00:00.000 -05:00, which is 18:54:48.8374563 from now.
[2024-08-19 07:05:11.382 -05:00] [INF] Found ffmpeg version "6.0.1"
[2024-08-19 07:05:11.473 -05:00] [INF] Available "decoders": ["libdav1d", "av1", "av1_cuvid", "av1_qsv", "h264", "h264_qsv", "h264_cuvid", "hevc", "hevc_qsv", "hevc_cuvid", "mpeg2video", "mpeg2_qsv", "mpeg2_cuvid", "mpeg4", "mpeg4_cuvid", "msmpeg4", "vc1_qsv", "vc1_cuvid", "vp8", "libvpx", "vp8_cuvid", "vp8_qsv", "vp9", "libvpx-vp9", "vp9_cuvid", "vp9_qsv", "aac", "ac3", "dca", "flac", "mp3", "truehd"]
[2024-08-19 07:05:11.519 -05:00] [INF] Available "encoders": ["libsvtav1", "av1_nvenc", "av1_qsv", "av1_amf", "av1_vaapi", "libx264", "h264_amf", "h264_nvenc", "h264_qsv", "h264_v4l2m2m", "h264_vaapi", "libx265", "hevc_amf", "hevc_nvenc", "hevc_qsv", "hevc_vaapi", "mjpeg_qsv", "mjpeg_vaapi", "mpeg4", "msmpeg4", "libvpx", "libvpx-vp9", "aac", "libfdk_aac", "ac3", "alac", "dca", "flac", "libmp3lame", "libopus", "truehd", "libvorbis", "srt"]
[2024-08-19 07:05:11.551 -05:00] [INF] Available filters: ["deinterlace_qsv", "deinterlace_vaapi", "hwupload_cuda", "hwupload_vaapi", "libplacebo", "overlay_opencl", "overlay_qsv", "overlay_vaapi", "overlay_vulkan", "overlay_cuda", "procamp_vaapi", "scale_cuda", "scale_opencl", "scale_qsv", "scale_vaapi", "scale_vulkan", "tonemap_cuda", "tonemap_opencl", "tonemap_vaapi", "vpp_qsv", "yadif_cuda", "zscale", "alphasrc"]
[2024-08-19 07:05:11.753 -05:00] [INF] Available hwaccel types: ["cuda", "vaapi", "qsv", "drm", "opencl", "vulkan"]
[2024-08-19 07:05:13.571 -05:00] [INF] FFmpeg: "/usr/lib/jellyfin-ffmpeg/ffmpeg"
[2024-08-19 07:05:13.573 -05:00] [INF] ServerId: "5392dc20830548669c10c5b6dcdd6648"
[2024-08-19 07:05:13.573 -05:00] [INF] Core startup complete
[2024-08-19 07:05:13.574 -05:00] [INF] Startup complete 0:00:22.2585412
[2024-08-19 07:05:14.199 -05:00] [INF] "Clean up collections and playlists" Completed after 0 minute(s) and 0 seconds
[2024-08-19 07:05:15.527 -05:00] [INF] "Update Plugins" Completed after 0 minute(s) and 1 seconds
[2024-08-19 07:05:32.356 -05:00] [INF] Watching directory "/mnt/media/Music/beet/music"
[2024-08-19 08:34:44.306 -05:00] [INF] "Download missing subtitles" Completed after 0 minute(s) and 0 seconds
[2024-08-19 08:34:44.343 -05:00] [INF] "Clean Transcode Directory" Completed after 0 minute(s) and 0 seconds
[2024-08-19 08:34:44.379 -05:00] [INF] "Clean Log Directory" Completed after 0 minute(s) and 0 seconds
[2024-08-19 08:34:44.445 -05:00] [INF] "TasksRefreshChannels" Completed after 0 minute(s) and 0 seconds
[2024-08-19 08:34:44.474 -05:00] [INF] Optimizing and vacuuming jellyfin.db...
[2024-08-19 08:34:44.492 -05:00] [INF] jellyfin.db optimized successfully!
[2024-08-19 08:34:44.492 -05:00] [INF] "Optimize database" Completed after 0 minute(s) and 0 seconds
[2024-08-19 08:34:45.325 -05:00] [INF] "Update Plugins" Completed after 0 minute(s) and 0 seconds
[2024-08-19 08:35:06.543 -05:00] [INF] Stopping directory watching for path "/mnt/media/Music/beet/music"
[2024-08-19 08:35:06.559 -05:00] [INF] Validating media library
[2024-08-19 08:35:06.802 -05:00] [WRN] Library folder "/var/lib/jellyfin/data/playlists" is inaccessible or empty, skipping
[2024-08-19 08:35:06.930 -05:00] [INF] Creating image collage and saving to "/var/cache/jellyfin/temp/28fec3e4cd334e488f16fe2411529dd5.png"
[2024-08-19 08:35:07.193 -05:00] [INF] Completed creation of image collage and saved to "/var/cache/jellyfin/temp/28fec3e4cd334e488f16fe2411529dd5.png"
[2024-08-19 08:35:07.338 -05:00] [WRN] Library folder "/var/lib/jellyfin/data/playlists" is inaccessible or empty, skipping
[2024-08-19 08:35:33.976 -05:00] [INF] "Scan Media Library" Completed after 0 minute(s) and 27 seconds
[2024-08-19 08:35:34.236 -05:00] [INF] Watching directory "/mnt/media/Music/beet/music"
[2024-08-19 09:06:50.334 -05:00] [INF] "Clean Cache Directory" Completed after 0 minute(s) and 1 seconds
[2024-08-19 09:07:25.069 -05:00] [INF] Refreshing guide with 7 days of guide data
[2024-08-19 09:07:25.078 -05:00] [INF] "Refresh Guide" Completed after 0 minute(s) and 3 seconds
[2024-08-19 12:40:29.004 -05:00] [INF] "Soundtracks" ("/mnt/media/Music/beet/music/Soundtracks") will be refreshed.
[2024-08-19 12:49:16.891 -05:00] [INF] "Soundtracks" ("/mnt/media/Music/beet/music/Soundtracks") will be refreshed.
[2024-08-19 12:51:25.348 -05:00] [INF] "Gimme All Your Lovin’" ("/mnt/media/Music/beet/music/ZZ Top/Greatest Hits/01 - Gimme All Your Lovin'.mp3") will be refreshed.
[2024-08-19 12:51:25.377 -05:00] [INF] Starting "/usr/lib/jellyfin-ffmpeg/ffprobe" with args "-analyzeduration 200M -probesize 1G -i file:\"/mnt/media/Music/beet/music/ZZ Top/Greatest Hits/01 - Gimme All Your Lovin'.mp3\" -threads 0 -v warning -print_format json -show_streams -show_format"
```

### Additional information

I know this worked in a previous version of Jellyfin, with this same beets library. IMO it seems like some kind of import behavior has changed.
I have also tried changing the genre format to `"Classic Rock\\Rock\\Southern Rock"` since this is used by some tag formats, but it still imported as a single string instead of separate entries.

**Comment by viown** [comment]
The genres need to be properly split in the metadata of the music files. Jellyfin no longer supports splitting by delimiters like commas or semicolons.

**Comment by xcorvis** [comment]
OK, I understand. It would be great if this were documented, I've searched a for this quite a bit and didn't find anything. I don't even see anything in the changelogs about it.

**Comment by gnattu** [comment]
You need to wait for https://github.com/jellyfin/jellyfin/pull/12385 which is currently targeting 10.10 to solve your problem.

Genres never supported comma supported comma separated value iirc and the implementation is linked PR is the only one that applies that to Generes.

**Comment by xcorvis** [comment]
> You need to wait for #12385 which is currently targeting 10.10 to solve your problem.
> 
> Genres never supported comma supported comma separated value iirc and the implementation is linked PR is the only one that applies that to Generes.

That's weird, because it definitely used to work this way. I'm even using the same beets library I used when it worked. In any case, thanks for pointing me at the right issue.

**Comment by xcorvis** [comment]
This is probably worthy of a separate ticket, but I rewrote all my genre tags to use a single item and it looks like Jellyfin isn't able to delete the old genres that contain commas. They won't delete, but If I edit the names of them to remove the commas they disappear.

**Comment by ssnarf** [comment]
> That's weird, because it definitely used to work this way. I'm even using the same beets library I used when it worked. In any case, thanks for pointing me at the right issue.

I learned today that there is a [long-standing issue in ffmpeg](https://trac.ffmpeg.org/ticket/6949) that prevents reading of multiple values in ID3v2.4 tags.

I had recently changed my tagging workflow and Picard was set to save as v2.4. Converting the tags to v2.3 solved the issue for me. Jellyfin now reads and separates delimited genre tags.

**Comment by gnattu** [comment]
> > That's weird, because it definitely used to work this way. I'm even using the same beets library I used when it worked. In any case, thanks for pointing me at the right issue.
> 
> I learned today that there is a [long-standing issue in ffmpeg](https://trac.ffmpeg.org/ticket/6949) that prevents reading of multiple values in ID3v2.4 tags.
> 
> I had recently changed my tagging workflow and Picard was set to save as v2.4. Converting the tags to v2.3 solved the issue for me. Jellyfin now reads and separates delimited genre tags.

Jellyfin 10.10 will no longer use ffmpeg as primary source and 10.9 already stopped using that as well and ffmpeg is now only used as a fallback. The problem is that the library 10.9 is using is quite limited and does have bugs that prevent certain mp3 tags being parsed correctly. You can try unstable to see if that already fixed your issue for id3v2.4

**Comment by richlawson** [comment]
10.10 fixed the issue for me. For my use case, re-scanning my music library and replacing all metadata did the trick.

**Comment by felix920506** [comment]
Closing as fixed in 10.10
